src/lib.rs

```diff
@@ -4,8 +4,17 @@ use ispc::WeightCollection;
 
 mod ispc;
 
+pub trait ImagePixelFormat {
+    fn num_channel_in_memory(&self) -> usize;
+    fn channel_size_in_bytes(&self) -> usize;
+
+    fn pixel_size(&self) -> usize {
+        self.channel_size_in_bytes() * self.num_channel_in_memory()
+    }
```

Comment:
Doc-comments? I'll continue reading the PR, but already have to confess that I don't understand what `num_channel_in_memory()` is supposed to return from just this definition.

Reply:
For API-cleanliness, maybe the trait coult require `: Copy`, and pass `self` by value.